yamux/src/connection/stream.rs

```diff
@@ -422,29 +609,143 @@ impl AsyncWrite for Stream {
     }
 }
 
+impl Drop for Stream {
+    fn drop(&mut self) {
+        let max_receive_window_size = self.shared.lock().max_receive_window_size;
+        let mut accumulated_max_stream_windows = self.accumulated_max_stream_windows.lock();
+
+        debug_assert!(
+            *accumulated_max_stream_windows >= (max_receive_window_size - DEFAULT_CREDIT) as usize,
+            "{accumulated_max_stream_windows} {max_receive_window_size}"
+        );
+
+        *accumulated_max_stream_windows -= (max_receive_window_size - DEFAULT_CREDIT) as usize;
+    }
+}
+
+#[cfg(test)]
+impl Clone for Stream {
+    fn clone(&self) -> Self {
+        Self {
+            id: self.id.clone(),
+            conn: self.conn.clone(),
+            config: self.config.clone(),
+            sender: self.sender.clone(),
+            flag: self.flag.clone(),
+            shared: Arc::new(Mutex::new(self.shared.lock().clone())),
+            accumulated_max_stream_windows: Arc::new(Mutex::new(
+                self.accumulated_max_stream_windows.lock().clone(),
+            )),
+            rtt: self.rtt.clone(),
+            last_window_update: self.last_window_update.clone(),
+        }
+    }
+}
+
+#[cfg(test)]
+impl quickcheck::Arbitrary for Stream {
```

Comment:
This looks like a massive hack for the benefit of calling `next_window_update` in a proptest. It would be better to extract everything around window-updates into a dedicated struct, test that using `proptest` and embed it within `Stream`.